DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -120,44 +119,31 @@ protected ProviderConfiguration ProviderConfiguration
 
         /// -----------------------------------------------------------------------------
         /// <summary>
-        /// Gets the Pre-Upgrade Script (if present) - this script will always run before
+        /// Gets a list of Pre-Upgrade Scripts (if present) - these scripts will always run before
         /// any upgrade scripts but not upon initial installation.
         /// </summary>
-        /// <value>An InstallFile.</value>
+        /// <value>A list of <see cref="InstallFile"/> instances.</value>
         /// -----------------------------------------------------------------------------
-        protected InstallFile PreUpgradeScript
+        protected List<InstallFile> PreUpgradeScripts
         {
             get
             {
-                return this._preUpgradeScript;
+                return this._preUpgradeScripts;
             }
         }
 
         /// -----------------------------------------------------------------------------
         /// <summary>
-        /// Gets the Upgrade Script (if present).
+        /// Gets a list of Post-Upgrade Scripts (if present) - these scripts will always run after
+        /// and versioned upgrade scripts and also after initial install.
         /// </summary>
-        /// <value>An InstallFile.</value>
+        /// <value>A list of <see cref="InstallFile"/> instances.</value>
         /// -----------------------------------------------------------------------------
-        protected InstallFile UpgradeScript

```

Comment:
Do we need to keep this around for backwards compatibility?  I'm not sure what the use case for extensing `ScriptInstaller` would be, but it's certainly possible…  Maybe have this return the first item in `PostUpgradeScripts`? 

```c#
protected InstallFile UpgradeScript => this.PostUpgradeScripts[0];
```

Reply:
Implemented